src/controllers/molecules/MessageComposer.js

```diff
@@ -19,10 +19,20 @@ limitations under the License.
 var MatrixClientPeg = require("../../MatrixClientPeg");
 
 var dis = require("../../dispatcher");
+var KeyCode = {
+    ENTER: 13,
+    TAB: 9,
+    SHIFT: 16
+};
 
 module.exports = {
     componentDidMount: function() {
         this.dispatcherRef = dis.register(this.onAction);
+        this.tabStruct = {
+            completing: false,
+            original: null,
+            index: 0
+        };
```

Comment:
In this case it probably doesn't really matter, but potentially better to put this in WillMount, then all your state is definitely set up before you do anything.